client/build.gradle.kts

```diff
@@ -32,8 +32,9 @@ val spineBaseVersion: String by extra
 dependencies {
     api(project(":core"))
 
-    Grpc.apply {
-        api(stub)
+    val deps = this
+    with(Grpc) {
+        deps.api(stub)
```

Comment:
This is an interesting one. So it's like in JS now? `this` references the closure?

Reply:
Apparently, it does :)

The problem here is that `Grpc` among its members has `api`. Which is coincidentally the same word as the Gradle's dependency container uses for its `api` configuration. Hence, this version does not work properly:
```kotlin

 with(Grpc) {
        api(stub)    // <- here is the clash of `api`s.
        implementation(protobuf)
        implementation(core)
    }

```

Therefore, I had to pick `this` from the first closure and remember it.

Reply:
See [This](https://kotlinlang.org/docs/this-expressions.html) explained.

I believe, we have the case as described:

> In an extension function or a function literal with receiver this denotes the receiver parameter that is passed on the left-hand side of a dot.

since `dependencies` _is_ an extension function:

```kotlin
fun Project.dependencies(configuration: DependencyHandlerScope.() -> Unit) =
    DependencyHandlerScope.of(dependencies).configuration()
```